src/charging-station/ChargingStationWorker.ts

```diff
@@ -28,11 +41,90 @@ if (ChargingStationUtils.workerPoolInUse()) {
   }
 }
 
+class TEMP {
+  public hashId: string;
+  public command: CommandCode;
+  public connectorId: number;
+  public idTag: string | null;
+}
+
+let station: ChargingStation;
+const channel = new BroadcastChannel('test');
```

Comment:
You need to: 

+ Encapsulate BC basic features (emit request, event handlers) into helpers that will open and close it if needed;  
+ BC shall be initialized on the main thread and pushed down to worker.

I do not see own you current code can work.